Bot crashes when activity in another channel

Crashes when activity on the discord channel (in another text channel).
Any recommendations on what to do? 

Latest error log attached. 
```

0 info it worked if it ends with ok
1 verbose cli [ '/root/.nvm/versions/node/v9.5.0/bin/node',
1 verbose cli   '/root/.nvm/versions/node/v9.5.0/bin/npm',
1 verbose cli   'start' ]
2 info using npm@5.6.0
3 info using node@v9.5.0
4 verbose run-script [ 'prestart', 'start', 'poststart' ]
5 info lifecycle messenger-discord@2.4.1~prestart: messenger-discord@2.4.1
6 info lifecycle messenger-discord@2.4.1~start: messenger-discord@2.4.1
7 verbose lifecycle messenger-discord@2.4.1~start: unsafe-perm in lifecycle true
8 verbose lifecycle messenger-discord@2.4.1~start: PATH: /root/.nvm/versions/node/v9.5.0/lib/node_modules/npm/node_modules/npm-lifecycle/node-gyp-bin:/home/stuff/messenger-discord/node_modules/.bin:/root/.nvm$
9 verbose lifecycle messenger-discord@2.4.1~start: CWD: /home/stuff/messenger-discord
10 silly lifecycle messenger-discord@2.4.1~start: Args: [ '-c', 'node index.js' ]
11 silly lifecycle messenger-discord@2.4.1~start: Returned: code: 1  signal: null
12 info lifecycle messenger-discord@2.4.1~start: Failed to exec start script
13 verbose stack Error: messenger-discord@2.4.1 start: `node index.js`
13 verbose stack Exit status 1
13 verbose stack     at EventEmitter.<anonymous> (/root/.nvm/versions/node/v9.5.0/lib/node_modules/npm/node_modules/npm-lifecycle/index.js:285:16)
13 verbose stack     at EventEmitter.emit (events.js:160:13)
13 verbose stack     at ChildProcess.<anonymous> (/root/.nvm/versions/node/v9.5.0/lib/node_modules/npm/node_modules/npm-lifecycle/lib/spawn.js:55:14)
13 verbose stack     at ChildProcess.emit (events.js:160:13)
13 verbose stack     at maybeClose (internal/child_process.js:943:16)
13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:220:5)
14 verbose pkgid messenger-discord@2.4.1
15 verbose cwd /home/stuff/messenger-discord
16 verbose Linux 4.4.0-116-generic
17 verbose argv "/root/.nvm/versions/node/v9.5.0/bin/node" "/root/.nvm/versions/node/v9.5.0/bin/npm" "start"
18 verbose node v9.5.0
19 verbose npm  v5.6.0
20 error code ELIFECYCLE
21 error errno 1
22 error messenger-discord@2.4.1 start: `node index.js`
22 error Exit status 1
23 error Failed at the messenger-discord@2.4.1 start script.
23 error This is probably not a problem with npm. There is likely additional logging output above.
24 verbose exit [ 1, true ]
```

Attaching also a text from crash. 

```
# npm start

> messenger-discord@2.4.1 start /home/zytriss/messenger-discord
> node index.js

info login Logged in
info login Request to reconnect
info login Request to pull 1
info login Request to pull 2
info login Request to thread_sync
info login Done logging in.
info listen Got answer in 918
info listen Got answer in 320
/home/stuff/messenger-discord/index.js:29
  if (message.channel.parent.name !== config.discord.category.name) return
                             ^

TypeError: Cannot read property 'name' of null
    at Client.discordListener (/home/stuff/messenger-discord/index.js:29:30)
    at Client.emit (events.js:160:13)
    at MessageCreateHandler.handle (/home/stuff/messenger-discord/node_modules/discord.js/src/client/websocket/packets/handlers/MessageCreate.js:9:34)
    at WebSocketPacketManager.handle (/home/stuff/messenger-discord/node_modules/discord.js/src/client/websocket/packets/WebSocketPacketManager.js:103:65)
    at WebSocketConnection.onPacket (/home/stuff/messenger-discord/node_modules/discord.js/src/client/websocket/WebSocketConnection.js:330:35)
    at WebSocketConnection.onMessage (/home/stuff/messenger-discord/node_modules/discord.js/src/client/websocket/WebSocketConnection.js:293:17)
    at WebSocket.onMessage (/home/stuff/messenger-discord/node_modules/ws/lib/event-target.js:120:16)
    at WebSocket.emit (events.js:160:13)
    at Receiver._receiver.onmessage (/home/stuff/messenger-discord/node_modules/ws/lib/websocket.js:145:47)
    at Receiver.dataMessage (/home/stuff/messenger-discord/node_modules/ws/lib/receiver.js:430:14)
    at Receiver.getData (/home/stuff/messenger-discord/node_modules/ws/lib/receiver.js:368:12)
    at Receiver.startLoop (/home/stuff/messenger-discord/node_modules/ws/lib/receiver.js:164:16)
    at Receiver.add (/home/stuff/messenger-discord/node_modules/ws/lib/receiver.js:138:10)
    at TLSSocket._ultron.on (/home/stuff/messenger-discord/node_modules/ws/lib/websocket.js:143:54)
    at TLSSocket.emit (events.js:160:13)
    at addChunk (_stream_readable.js:269:12)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! messenger-discord@2.4.1 start: `node index.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the messenger-discord@2.4.1 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2018-02-22T14_19_47_938Z-debug.log


```

--- a/index.js
+++ b/index.js
@@ -26,7 +26,7 @@ function discordListener (message) {
   if (!parseInt(message.channel.topic, 10).toString() === message.channel.topic) return
 
   // make sure it's bot's category
-  if (message.channel.parent.name !== config.discord.category.name) return
+  if (message.channel.parent && message.channel.parent.name !== config.discord.category.name) return
 
   // build message with attachments provided
   var msg = {

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "messenger-discord",
-  "version": "2.4.1",
+  "version": "2.4.2",
   "description": "Facebook Messenger to Discord bridge",
   "main": "index.js",
   "scripts": {

